vms/avm/tx.go

```diff
@@ -0,0 +1,134 @@
+// Copyright (C) 2019-2023, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package avm
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"go.uber.org/zap"
+
+	"github.com/ava-labs/avalanchego/database"
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/ava-labs/avalanchego/snow/choices"
+	"github.com/ava-labs/avalanchego/snow/consensus/snowstorm"
+	"github.com/ava-labs/avalanchego/utils/set"
+	"github.com/ava-labs/avalanchego/vms/avm/txs"
+	"github.com/ava-labs/avalanchego/vms/avm/txs/executor"
+)
+
+var (
+	_ snowstorm.Tx = (*Tx)(nil)
+
+	errTxNotProcessing  = errors.New("transaction is not processing")
+	errUnexpectedReject = errors.New("attempting to reject transaction")
+)
+
+type Tx struct {
+	vm *VM
+	tx *txs.Tx
+}
+
+func (tx *Tx) ID() ids.ID {
+	return tx.tx.ID()
+}
+
+func (tx *Tx) Accept(context.Context) error {
+	if s := tx.Status(); s != choices.Processing {
+		return fmt.Errorf("%w: %s", errTxNotProcessing, s)
+	}
+
+	if err := tx.vm.onAccept(tx.tx); err != nil {
+		return err
+	}
+
+	executor := &executor.Executor{
+		Codec: tx.vm.txBackend.Codec,
+		State: tx.vm.state,
+		Tx:    tx.tx,
+	}
+	err := tx.tx.Unsigned.Visit(executor)
+	if err != nil {
+		return fmt.Errorf("error staging accepted state changes: %w", err)
+	}
+
+	tx.vm.state.AddTx(tx.tx)
+
+	txID := tx.ID()
+	tx.vm.state.DeleteStatus(txID)
```

Comment:
We assume that transactions on disk with no status are accepted, so we just need to make sure to delete any previously existing statuses on disk.